osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -35,29 +43,16 @@ public class LegacyBeatmapDecoder : LegacyDecoder
 
         private readonly int offset = UniversalOffset;
 
-        public LegacyBeatmapDecoder()
-        {
-        }
-
-        public LegacyBeatmapDecoder(string header)
+        public LegacyBeatmapDecoder(int version = LATEST_VERSION) : base(version)
         {
-            BeatmapVersion = int.Parse(header.Substring(17));
-
             // BeatmapVersion 4 and lower had an incorrect offset (stable has this set as 24ms off)
-            offset += BeatmapVersion < 5 ? 24 : 0;
+            offset += FormatVersion < 5 ? 24 : 0;
         }
 
-        protected override void ParseBeatmap(StreamReader stream, Beatmap beatmap)
+        protected override void ParseStreamInto(StreamReader stream, Beatmap beatmap)
         {
-            if (stream == null)
-                throw new ArgumentNullException(nameof(stream));
-            if (beatmap == null)
-                throw new ArgumentNullException(nameof(beatmap));
-
             this.beatmap = beatmap;
-            this.beatmap.BeatmapInfo.BeatmapVersion = BeatmapVersion;

```

Comment:
This line was removed. Is BeatmapVersion still getting correctly set somewhere else? It's still being used in cases like mania and taiko conversion.

Reply:
Looks like a regression, good catch.